### DISCUSSION TITLE
Allow to run migrations manually contribution

### DISCUSSION THREAD

**Discussion by shawnest** [jobrunr/jobrunr#372] [Q&A]
I'm planning to contribute to the project with issue: https://github.com/jobrunr/jobrunr/issues/356.
Could somebody guide me to the right direction?

I was wondering which class should expose the migration API. I'm thinking about putting it into the various StorageProvider classes (DefaultSQLStorageProvider and the NoSQL providers).

What should I test and what kind of tests are needed to submit the pull request?

**Comment by rdehuyss** [comment]
Hi @shawnest ,

Thanks for your support and willingness to contribute.

I don't know if there is much to test honestly - all the different components are already tested so that should be ok in my eyes.

I would propose the following:
- A new method is added on the `StorageProvider` interface called `setUpStorageProvider()`
- this method is than implemented for both the Sql and NoSql StorageProviders
- the `DefaultSqlStorageProvider` would then become something like (we would also not need the field DatabaseOption anymore):

```java
public void setUpStorageProvider() {
    this.createDBIfNecessary(CREATE);
}

protected void createDBIfNecessary(DatabaseOption databaseOption) {
    if (databaseOptions == CREATE) {
        getDatabaseCreator()
                    .runMigrations();
    } else {
        getDatabaseCreator()
                    .validateTables();
    }
}
```

The main goal would be to make it backwards compatible. 

Please also base yourself on the feature/v5 branch.

I hope this helps and thanks!
